client/src/context.tsx

```diff
@@ -82,6 +84,27 @@ const reducer = (state: State, action: Action) => {
         }
       }
     }
+    case 'DELETE_SLIDE': {
+      const { slideID } = action.payload
+
+      const user = state.auth.user
+
+      if (!user) {
+        return state
+      }
+
+      const { slides } = user
+
+      const newSlides = slides.filter(el => el._id != slideID)
+
+      return {
+        ...state,
+        user: {
+          ...user,
+          slides: [...newSlides]
+        }
+      }
+    }
```

Comment:
The return statement is missing the auth which has the user. I guess this is why it's not working for you.